src/Microsoft.ML.Api/InternalSchemaDefinition.cs

```diff
@@ -131,19 +130,29 @@ private InternalSchemaDefinition(Column[] columns)
         }
 
         /// <summary>
-        /// Given a field info on a type, returns whether this appears to be a vector type,
+        /// Given a field or property info on a type, returns whether this appears to be a vector type,
         /// and also the associated data kind for this type. If a data kind could not
         /// be determined, this will throw.
         /// </summary>
-        /// <param name="fieldInfo">The field info to inspect.</param>
+        /// <param name="memberInfo">The field or property info to inspect.</param>
         /// <param name="isVector">Whether this appears to be a vector type.</param>
         /// <param name="kind">The data kind of the type, or items of this type if vector.</param>
-        public static void GetVectorAndKind(FieldInfo fieldInfo, out bool isVector, out DataKind kind)
+        public static void GetVectorAndKind(MemberInfo memberInfo, out bool isVector, out DataKind kind)
         {
-            Contracts.AssertValue(fieldInfo);
-            Type rawFieldType = fieldInfo.FieldType;
-            var name = fieldInfo.Name;
-            GetVectorAndKind(rawFieldType, name, out isVector, out kind);
+            Contracts.AssertValue(memberInfo);
+            switch (memberInfo)
+            {
+                case FieldInfo fieldInfo:
+                    GetVectorAndKind(fieldInfo.FieldType, fieldInfo.Name, out isVector, out kind);
+                    break;
+
+                case PropertyInfo propertyInfo:
+                    GetVectorAndKind(propertyInfo.PropertyType, propertyInfo.Name, out isVector, out kind);
+                    break;
+
+                default:
+                    throw Contracts.ExceptNotSupp("expected a FieldInfo or a PropInfo");
```

Comment:
>PropInfo [](start = 77, length = 8)

is it worth to be shorten?
Also can you start sentence with capital letter?

Reply:
(nit) `PropInfo` => `PropertyInfo`

---
In reply to: [206681072](https://github.com/dotnet/machinelearning/pull/616#discussion_r206681072) [](ancestors = 206681072)